var/spack/repos/builtin/packages/py-pre-commit/package.py

```diff
@@ -13,18 +13,22 @@ class PyPreCommit(PythonPackage):
     homepage = "https://github.com/pre-commit/pre-commit"
     pypi = "pre_commit/pre_commit-1.20.0.tar.gz"
 
+    version('2.17.0', sha256='c1a8040ff15ad3d648c70cc3e55b93e4d2d5b687320955505587fd79bbaed06a')
     version('1.20.0', sha256='9f152687127ec90642a2cc3e4d9e1e6240c4eb153615cb02aa1ad41d331cbb6e')
 
     depends_on('python@2.7:2.8,3.5:', type=('build', 'run'))
+    depends_on('python@3.6.1:', type=('build', 'run'), when="@2.1.0:")
     depends_on('py-setuptools', type='build')
-    depends_on('py-aspy-yaml', type=('build', 'run'))
+    depends_on('py-aspy-yaml', type=('build', 'run'), when="@:2")
     depends_on('py-cfgv@2.0.0:', type=('build', 'run'))
     depends_on('py-identify@1.0.0:', type=('build', 'run'))
     depends_on('py-nodeenv@0.11.1:', type=('build', 'run'))
     depends_on('py-pyyaml', type=('build', 'run'))
-    depends_on('py-six', type=('build', 'run'))
+    depends_on('py-pyyaml@5.1:', type=('build', 'run'), when="@2.1.0:")
+    depends_on('py-six', type=('build', 'run'), when="@:2")
```

Comment:
`@:2` means all versions up to and including 2.X.Y. Maybe you want `@1` (all 1.X.Y versions) or `@:2.1` (all versions up to and including 2.1.Y?

Reply:
I'll fix this shortly